Improved Provider and Model Name Handling

NANDINI-star opened the pull request:
## Title
Don't require user to set the provider name when writing a custom model name

<!-- e.g. "Implement user authentication feature" -->
<!-- e.g. "Fixes #000" -->

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🆕 Enhancement

## Changes
The changes include sorting provider names, automatically adding model names, and prepending the provider to the selected model name.
Key Changes:
- Sorted Provider Names: The list of providers is now alphabetically sorted in the "Add New Model" form, making it easier for users to find the provider they're looking for.
- Automatic Model Name Population: When a user selects a provider, the associated model names are now automatically populated in the model name fields. 
- Provider Name Prefix: For clarity, the selected provider's name is now automatically prepended to the model name. 

https://www.loom.com/share/b33d22a4217c4edebb21241596e99272?sid=25e3f123-0008-48cf-b812-7502368661a0

NANDINI-star committed:
sort the provider names

NANDINI-star committed:
add / and name for selected models

NANDINI-star committed:
autoadd custom model names

NANDINI-star committed:
remove unused imports+ prettier

NANDINI-star committed:
minor

NANDINI-star committed:
Merge branch 'main' into provider-name-preset

krrishdholakia commented:
<img width="649" height="240" alt="Screenshot 2025-07-14 at 10 32 24 PM" src="https://github.com/user-attachments/assets/c02647bc-c710-4dab-b52e-3a3e0cc06a09" />


the `anthropic` part of this should be non-editable. the user can add additional input after `anthropic/` - e.g. `anthropic/claude-3-5-sonnet`, so they can specify the model they want to call.

krrishdholakia commented:
it should also be clear what the 'anthropic/` is

You could just have the `anthropic/` part show up under litellm model name in the table, since it's not editable 

<img width="318" height="179" alt="Screenshot 2025-07-14 at 10 33 54 PM" src="https://github.com/user-attachments/assets/dbc86caa-7bbd-4648-86d5-1b93f23716ca" />

NANDINI-star committed:
add "/" to litellm model

NANDINI-star committed:
remove unrequired my-deployment

NANDINI-star committed:
Merge branch 'main' into provider-name-preset

krrishdholakia commented:
<img width="1015" height="223" alt="Screenshot 2025-08-12 at 5 24 41 PM" src="https://github.com/user-attachments/assets/7c6de0aa-466a-4601-bc5a-dcfd23126a7e" />


@NANDINI-star this doesn't solve the problem because i need to edit the litellm model name, to enter my azure deployment name 

the ask was to not require me to remember to add `azure/` before entering my deployment name, since you already know the provider

NANDINI-star commented:
> @NANDINI-star this doesn't solve the problem because i need to edit the litellm model name, to enter my azure deployment name
the ask was to not require me to remember to add azure/ before entering my deployment name, since you already know the provider

@krrishdholakia 

https://www.loom.com/share/b33d22a4217c4edebb21241596e99272?sid=25e3f123-0008-48cf-b812-7502368661a0

This is not the latest loom since it was made before addressing the review comments, but looking at 00:07 is that what you mean?

krrishdholakia commented:
this looks buggy @NANDINI-star 

look at the malformed public model name 

<img width="619" height="288" alt="Screenshot 2025-08-12 at 5 44 44 PM" src="https://github.com/user-attachments/assets/057843f3-1a61-4a12-bac7-f1ef45d77ea1" />

---

also, my comment is still not addressed - i just want to enter my azure deployment name, not `azure/` - which is what you **need** to do for the model routing to work on litellm - https://docs.litellm.ai/docs/providers/azure#completion---using-env-variables

as the ui already knows the provider, i want you to prepend `azure/` to the litellm model name, without me having to remember to do that.

NANDINI-star committed:
Merge branch 'main' into provider-name-preset
